luarules/gadgets/map_nightmode.lua

```diff
@@ -421,6 +421,15 @@ if not gadgetHandler:IsSyncedCode() then
 		lastSunChanged = df
 	end
 
+	local function isAuthorizedWrapper(func)
```

Comment:
would be nice if this was some sort of `GG.AuthorizedChatActionWrapper` in dev helpers, since it's going to look the same in other gadgets

Reply:
ChatAction always has playerID as the 4th arg (apparently) so i think you could do
```lua
function GG.AuthorizedChatActionWrapper(func)
  return function(a, b, c, playerID, ...)
    if allowed(playerID) then
      func(a, b, c, playerID, ...)
    end
  end
end
```